pkg/apis/openstack/validation/shoot.go

```diff
@@ -60,9 +72,46 @@ func ValidateWorkersUpdate(oldWorkers, newWorkers []core.Worker, fldPath *field.
 				if validation.ShouldEnforceImmutability(newWorker.Zones, oldWorker.Zones) {
 					allErrs = append(allErrs, apivalidation.ValidateImmutableField(newWorker.Zones, oldWorker.Zones, fldPath.Index(i).Child("zones"))...)
 				}
-				break
+
+				if newWorker.ProviderConfig != nil && oldWorker.ProviderConfig != nil {
+					allErrs = append(allErrs, apivalidation.ValidateImmutableField(string(newWorker.ProviderConfig.Raw), string(oldWorker.ProviderConfig.Raw), fldPath.Index(i).Child("providerConfig"))...)
+				} else if newWorker.ProviderConfig != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("providerConfig"), string(newWorker.ProviderConfig.Raw), apivalidation.FieldImmutableErrorMsg))
+				} else if oldWorker.ProviderConfig != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("providerConfig"), nil, apivalidation.FieldImmutableErrorMsg))
```

Comment:
The whole provider config is immutable? This makes it hard to extend it later. Probably it's better to just make the fields inside the provider config immutable.

Reply:
There was no provider config before this change, thus making it immutable is semantically equivalent to making the fields immutable. I opted for this, so as to not introduce complexity which is not needed immediately. But yes, should we need to extend the config, then we would need to make the fields immutable and I can do that now.

Reply:
Yes, let's do it now while we are on it anyways, this makes further contributions easier!